test/parallel/test-util-inspect-long-running.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+const common = require('../common');
+
+// `util.inspect` is capable of computing a string that is bigger than 2 ** 28
+// in one second, so let's just skip this test on 32bit systems.
+common.skipIf32Bits();
+
+// Test that big objects are running only up to the maximum complexity plus ~1
+// second.
+
+const util = require('util');
+
+// Create a difficult to stringify object. Without the limit this would crash.
+let last = {};
+const obj = last;
+
+for (let i = 0; i < 1000; i++) {
+  last.next = { circular: obj, last, obj: { a: 1, b: 2, c: true } };
+  last = last.next;
+  obj[i] = last;
+}
+
+common.expectWarning(
+  'Warning',
+  'util.inspect took to long.',
+  'INSPECTION_ABORTED'
+);
+
+util.inspect(obj, { depth: Infinity });
```

Comment:
Can we add tests that test the limits here?

I think we need a test for the complexity parameter to make sure it works

Reply:
It is tested implicitly here. Without the option being set the test would crash.

Reply:
@BridgeAR I realize, but for example if we break the code that parses the option then tests wouldn't catch it - so if it's not a lot of trouble I think adding a second test that sets the configuration explicitly is a good idea.

<sub>Definitely not blocking on this though.</sub>